### PR TITLE
go/proxyd: Add request/response payload size metrics

### PR DESCRIPTION
**Pull request by Inphi**
**Description**
Request and response payload lengths are recorded as histograms.

**Metadata**
- Fixes ENG-1687

### CODE REVIEW

File: go/proxyd/metrics.go [hunk 1]
```diff
@@ -20,6 +21,8 @@ const (
 	MethodUnknown = "unknown"
 )
 
+var PayloadSizeBuckets = []float64{10, 50, 100, 500, 1000, 5000, 10000, 100000, 1000000}
```
Comment (Inphi) [inline go/proxyd/metrics.go:24]: @mslipper these buckets should suffice. lmkwyt

Comment (mslipper) [reply go/proxyd/metrics.go:24]: Yea they LGTM

File: go/proxyd/server.go [hunk 2]
```diff
@@ -291,3 +296,25 @@ func GetXForwardedFor(ctx context.Context) string {
 	}
 	return xff
 }
+
+type recordLenReader struct {
```
Comment (mslipper) [inline go/proxyd/server.go:300]: This is a neat solution!

**Review by mslipper — APPROVED** [review]

**Comment by mslipper** [comment]
Please add a changeset and I'll get this merged.

**Pull request merged**